### PR TITLE
Non-cat changes to master from cat branch, 1

### PR DESCRIPTION
**Pull request by rwest**
### Motivation or Problem
This is a collection of commits that prepares the groundwork for merging the cat branch (heterogeneous catalysis). I think they don't have anything to do with heterogenous catalysis.
My hope of collecting them in a separate pull request is that it will make code review easier.
We could split this PR further if desired.
I am hoping that these will not be too controversial or at least will not break anything.

### Description of Changes

So far...

- Fix a bug in the VF2 isomorphism code for disjoint graphs (will later be used for things physisorbed onto the surface)
- Improve error logging in various places (implemented while debugging over the last 2.5 years)
- Tweak a few docstrings and comments.
- Edits to chemkin writing. Using a new function to convert units (because surface reactions can have very different dimensions, and simple heuristics will fail).

### Testing

This now passes all the automated tests.

I believe the codecov report is not getting all the cythonized stuff.

### CODE REVIEW

**Comment by rwest** [comment]
Running locally I can reproduce the `continuous-integration/rmg-tests` failure only when I run with the `master` branch of the RMG-database. When I run with the `units` branch (which has some corrections to the database), as I tried to configured in https://github.com/ReactionMechanismGenerator/RMG-Py/pull/1538/commits/e074e0b0b90a811672ade57d39ac1a70d15e8756, it works.

I.e. I think https://github.com/ReactionMechanismGenerator/RMG-Py/pull/1538/commits/e074e0b0b90a811672ade57d39ac1a70d15e8756, which attempted to follow the instructions at https://github.com/ReactionMechanismGenerator/RMG-Py/wiki/Simultaneous-Update-of-RMG-Py-and-RMG-database, does not work for the continuous-integration/rmg-tests (although it does work for the continuous-integration/travis-ci/pr  and continuous-integration/travis-ci/push tests)

Perhaps once https://github.com/ReactionMechanismGenerator/RMG-database/pull/311 is merged to RMG-database, these tests can be re-run, and this PR reviewed and merged.

**Comment by rwest** [comment]
The units fix has been merged to the master branch of RMG-database, so I've removed the commit that changed database branch, and am hoping this PR now passes the tests...

File: rmgpy/molecule/molecule.py [hunk 1]
```diff
@@ -1001,6 +1001,18 @@ def getNumAtoms(self, element = None):
                     numAtoms += 1
             return numAtoms
 
+    def getNumberOfRadicalElectrons(self):
```
[the remark this answers is not in the document]
Comment (rwest) [reply rmgpy/molecule/molecule.py]: Yes, you removed it from master in 60bbcfdc626a8b6419a50788fb8baad869d6c735 but that got lost in a rebase, and it ended up being added again here (just by virtue of not being removed). Will remove again.

**Comment by rwest** [comment]
I've made the changes suggested by @mliu49 and rebased.

**Comment by mliu49** [comment]
@rwest, do you think you could also add the unit checking database test to the kinetics depositories tests? I found a unit mistake which was picked up by RMG-tests, and I expect there might be others as well.
```
R_Addition_MultipleBond/training

entry(
    index = 2,
    label = "CH2O + C3H5O <=> C4H7O2",
    degeneracy = 1.0,
    kinetics = Arrhenius(
        A = (54.3214, 's^-1', '*|/', 1.1507),
        n = 3.00879,
        Ea = (6.589, 'kcal/mol', '+|-', 0.024),
        T0 = (1, 'K'),
    ),
    referenceType = "theory",
    rank = 10,
    shortDesc = u"""nyee TST calculations at CBS-QB3 level with hindered rotors level""",
    longDesc = 
u"""
Quantum chemistry calculations at the CBS-QB3 level with hindered rotors
using Gaussian 03. High-pressure-limit rate coefficient computed
using Cantherm. One of the rotors had coupling and did not converge back to the
same initial geometry. It was forced to go back by editing the scan log.
""",
)
```

**Comment by rwest** [comment]
Good idea. Happy for you to add that if you have time before I do.

**Comment by mliu49** [comment]
Ok, I can do that this afternoon.

**Comment by mliu49** [comment]
@rwest, could you rebase this branch? I think it's ready to merge.

I can also do it if you're busy, but I don't want to mess up the other cat branches.

**Pull request merged**